bitnami/spark/values-production.yaml

```diff
@@ -0,0 +1,237 @@
+  readinessProbe:
+    enabled: true
+    initialDelaySeconds: 30
+    periodSeconds: 10
+    timeoutSeconds: 5
+    failureThreshold: 6
+    successThreshold: 1
+
+worker:
+  ## Spark worker specific configuration
+
+  webPort: 8081
+  ## Set to true to use a custom cluster port instead of a random port.
+  customClusterPort: false
+  # clusterPort:
+
+  ## Set the daemonMemoryLimit as the daemon max memory
+  # daemonMemoryLimit:
+
+  ## Set the worker memory limit
+  # memoryLimit:
+
+  ## Set the maximun number of cores
+  # coreLimit:
+
+  ## Working directory for the application
+  # dir:
+
+  ## Options for the JVM as "-Dx=y"
+  # javaOptions:
+
+  ## Configuraion options in the form "-Dx=y"
+  # configOptions:
+
+  ## Number of spark workers (will be the min number when autoscaling is enabled)
+  replicaCount: 2
+  ## Enable replica autoscaling depending on CPU
+  enableAutoscaling: false
+  ## Max number of workers when using autoscaling
+  # replicasMax:
+
+  ## Set to true if you would like to see extra information on logs
+  ## It turns BASH and NAMI debugging in minideb
+  ## ref:  https://github.com/bitnami/minideb-extras/#turn-on-bash-debugging
+  debug: false
+  ## StatefulSet controller supports automated updates. There are two valid update strategies: RollingUpdate and OnDelete
+  ## ref: https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#updating-statefulsets
+  ##
+  updateStrategy: RollingUpdate
+  ## Kubernetes Security Context
+  ## https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+  ##
+  securityContext:
+    enabled: true
+    fsGroup: 1001
+    runAsUser: 1001
+  ## Node labels for pod assignment
+  ## Ref: https://kubernetes.io/docs/user-guide/node-selection/
+  ##
+  nodeSelector: {}
+  ## Tolerations for pod assignment
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+  ##
+  tolerations: []
+  ## Affinity for pod assignment
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+  ##
+  affinity: {}
+  ## Configure resource requests and limits
+  ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+  ##
+  resources:
+  #  limits:
+  #    cpu: 200m
+  #    memory: 1Gi
+  #  requests:
+  #    memory: 256Mi
+  #    cpu: 250m
+
+  ## Configure extra options for liveness and readiness probes
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes)
+  livenessProbe:
+    enabled: true
+    initialDelaySeconds: 180
+    periodSeconds: 20
+    timeoutSeconds: 5
+    failureThreshold: 6
+    successThreshold: 1
+
+  readinessProbe:
+    enabled: true
+    initialDelaySeconds: 30
+    periodSeconds: 10
+    timeoutSeconds: 5
+    failureThreshold: 6
+    successThreshold: 1
+
+## Security configuration
+security:
+  ## Name of the secret that contains all the passwords.
+  passwordsSecretName: my-passwords-secret
```

Comment:
This can be confusing. I think that we should use something like the forcePassword feature in MariaDB

https://github.com/helm/charts/blob/master/stable/mariadb/templates/secrets.yaml#L15

Reply:
It is more less the same, the secret is to provide custom passwords, if the user do not create one, a secret with random passwords are automatically generated by the `secret.yaml` template.